### PR TITLE
Dyno: Default initializers for classes that inherit

### PR DESCRIPTION
**Pull request by benharsh**
This PR changes the frontend to start generating uAST for default initializers, which allows us to support default initializers on classes that inherit. 

This PR replaces the current signature-based implementation for default initializers with uAST for a Function. This Function's body contains the suitable statements to initialize each field, as well as a "super.init" call if applicable. This code currently lacks suitable init-expressions for default-initializable formals, which is left as future work.

An UntypedFnSignature is built manually so that we can force resolution to accept non-existent default values for each formal, regardless of whether it has an init-expression or not.

The resulting TypedFnSignature is created with `typedSignatureInitial`. This might be overkill, but it keeps the code cleaner for the time being.

-----

Now that default initializers have actual bodies, the frontend is attempting to resolve various assignment operators only defined in the internal/standard modules. This was initially a problem for the dedicated frontend tests, which do not generally load the internal modules in which basic assignment operators are implemented. This PR updates the frontend to generate assignment operators for primitive types, tuples, and classes when the standard library is not available.

Testing:
- [x] test-dyno

### CODE REVIEW

**Comment by benharsh** [comment]
Note for reviewer: this might be easier to review commit-by-commit

File: frontend/lib/resolution/default-functions.cpp [hunk 1]
```diff
@@ -293,40 +294,209 @@ static void buildInitArgs(Context* context,
   }
 }
 
+static void collectFields(const AstNode* ast,
+                          std::vector<const VarLikeDecl*>& fields) {
+  if (auto decl = ast->toAggregateDecl()) {
```
Comment (mppf) [inline frontend/lib/resolution/default-functions.cpp:299]: I suspect that this won't do what you want with a record containing a nested record. (Would that pattern get this far?)

**Review by mppf — APPROVED** [review]

File: frontend/lib/resolution/default-functions.cpp (hunk 1 above)
Comment (benharsh) [reply frontend/lib/resolution/default-functions.cpp]: Good catch! Fixed.

**Comment by bradcray** [comment]
Nice!

**Pull request merged**